packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/ApiKeyAuthImpl.kt

```diff
@@ -0,0 +1,124 @@
+package io.realm.kotlin.mongodb.internal
+
+import io.realm.kotlin.internal.ObjectIdImpl
+import io.realm.kotlin.internal.interop.ObjectIdWrapper
+import io.realm.kotlin.internal.interop.RealmInterop
+import io.realm.kotlin.internal.interop.sync.ApiKeyWrapper
+import io.realm.kotlin.internal.platform.freeze
+import io.realm.kotlin.internal.util.use
+import io.realm.kotlin.mongodb.auth.ApiKey
+import io.realm.kotlin.mongodb.auth.ApiKeyAuth
+import io.realm.kotlin.types.ObjectId
+import kotlinx.coroutines.channels.Channel
+
+internal class ApiKeyAuthImpl(val app: AppImpl, val user: UserImpl) : ApiKeyAuth {
+
+    override suspend fun create(name: String): ApiKey {
+        Channel<Result<ApiKey>>(1).use { channel ->
+            RealmInterop.realm_app_user_apikey_provider_client_create_apikey(
+                app.nativePointer,
+                user.nativePointer,
+                name,
+                channelResultCallback<ApiKeyWrapper, ApiKey>(channel) { apiKeyData ->
+                    ApiKey(
```

Comment:
I would consider moving this mapping into a helper method since it is shared between 3 methods and it is easy to get the boolean logic wrong.